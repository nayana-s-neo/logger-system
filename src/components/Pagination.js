import React, { useEffect, useState } from 'react';

function Pagination({recordCount,perPage,getRecords}) {
  const [pages, setPages] = useState([]);
  const [activePage, setActivePage] = useState(1);
  let remaining = parseInt(recordCount%perPage) > 0 ? 1 : 0;
  useEffect(() => {
    setPages( Array.from({length: parseInt(recordCount/perPage) + remaining}, (_, i) => i + 1));
  },[recordCount])

  const displayRecords = (page) => {
    getRecords(page);
    setActivePage(page);
  }
  return (
    <>
        {
            pages.map((page) => {
                let activeClass  = "";
                if(page == activePage)
                    activeClass = "activePage";
                return(
                    <span data-testid={`page-test-id${page}`} className={`pageNum ${activeClass}`} onClick = { () => displayRecords(page)} >{page}</span>
                )
            })
        }
    </>
  )
}

export default Pagination