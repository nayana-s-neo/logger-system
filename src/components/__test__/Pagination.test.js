import { render } from '@testing-library/react';
import {BrowserRouter as Router} from 'react-router-dom';
import Pagination from '../Pagination'


const displayRecords = jest.fn();
describe('pagination section',() => {
    
it('renders pagination section', () => {
  const {container} = render(<Router><Pagination recordCount={100} perPage={10}/></Router>);
  expect(container).toMatchSnapshot();
});

it('check no of pages present as well as not there',  () => {
    const {container, getByTestId,queryByTestId} = render(<Router><Pagination pages={10} recordCount={20} perPage={10} getRecords={jest.fn()}/></Router>);
    expect(getByTestId('page-test-id1')).toBeInTheDocument();
    expect(getByTestId('page-test-id2')).toBeInTheDocument();
    expect(queryByTestId('page-test-id3')).not.toBeInTheDocument();
})

});