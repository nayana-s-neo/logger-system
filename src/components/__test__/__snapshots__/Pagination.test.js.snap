// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination section renders pagination section 1`] = `
<div>
  <span
    class="pageNum activePage"
    data-testid="page-test-id1"
  >
    1
  </span>
  <span
    class="pageNum "
    data-testid="page-test-id2"
  >
    2
  </span>
  <span
    class="pageNum "
    data-testid="page-test-id3"
  >
    3
  </span>
  <span
    class="pageNum "
    data-testid="page-test-id4"
  >
    4
  </span>
  <span
    class="pageNum "
    data-testid="page-test-id5"
  >
    5
  </span>
  <span
    class="pageNum "
    data-testid="page-test-id6"
  >
    6
  </span>
  <span
    class="pageNum "
    data-testid="page-test-id7"
  >
    7
  </span>
  <span
    class="pageNum "
    data-testid="page-test-id8"
  >
    8
  </span>
  <span
    class="pageNum "
    data-testid="page-test-id9"
  >
    9
  </span>
  <span
    class="pageNum "
    data-testid="page-test-id10"
  >
    10
  </span>
</div>
`;
